File: AlgoPersonalStudy/src/backjoon/silver/BJ_1822.java
package backjoon.silver;

import java.io.*;
import java.util.*;

public class BJ_1822 {
    public static void main(String[] args) throws Exception {
        solution();
    }
    static void solution() throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int A = Integer.parseInt(st.nextToken());
        int B = Integer.parseInt(st.nextToken());

        HashSet<Long> hashSet = new HashSet<>();

        st = new StringTokenizer(br.readLine());
        for(int a=0;a<A;a++){
            long num = Long.parseLong(st.nextToken());
            hashSet.add(num);
        }

        st = new StringTokenizer(br.readLine());
        for(int b=0;b<B;b++){
            long num = Long.parseLong(st.nextToken());
            hashSet.remove(num);
        }
        if(hashSet.isEmpty()){
            System.out.println(0);
        }
        else {
            System.out.println(hashSet.size());
            ArrayList<Long> list = new ArrayList<>(hashSet);
            Collections.sort(list);

            StringBuilder sb = new StringBuilder();
            for (long num : list) {
                sb.append(num).append(" ");
            }
            System.out.println(sb.toString());
        }
    }
}

File: AlgoPersonalStudy/src/_kotlin/backjoon/silver/BJ_1822.kt
package _kotlin.backjoon.silver

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

class BJ_1822 {
    fun solution(){
        val br = BufferedReader(InputStreamReader(System.`in`))
        val (a, b) = br.readLine().split(" ").map { it.toInt() }
        val set = HashSet<Long>()

        var st = StringTokenizer(br.readLine())
        while(st.hasMoreTokens()){
            set.add(st.nextToken().toLong())
        }
        st = StringTokenizer(br.readLine())
        while(st.hasMoreTokens()){
            set.remove(st.nextToken().toLong())
        }
        if(set.isEmpty()) print(0)
        else{
            println(set.size)
            set.sorted()
            val sb = StringBuilder()
            for(num: Long in set){
                sb.append(num).append(" ")
            }
            println(sb.toString())
        }
    }
    fun solutionFromJava(){
        val br = BufferedReader(InputStreamReader(System.`in`))
        var st = StringTokenizer(br.readLine())
        val A = st.nextToken().toInt()
        val B = st.nextToken().toInt()

        val hashSet = HashSet<Long>()

        st = StringTokenizer(br.readLine())
        for (a in 0 until A) {
            val num = st.nextToken().toLong()
            hashSet.add(num)
        }

        st = StringTokenizer(br.readLine())
        for (b in 0 until B) {
            val num = st.nextToken().toLong()
            hashSet.remove(num)
        }
        if (hashSet.isEmpty()) {
            println(0)
        } else {
            println(hashSet.size)
            val list = ArrayList(hashSet)
            list.sort()
            val sb = java.lang.StringBuilder()
            for (num in list) {
                sb.append(num).append(" ")
            }
            println(sb.toString())
        }
    }
}
fun main(){
    val solutionClass = BJ_1822()
    solutionClass.solution()
}